pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -112,28 +114,60 @@ func (d *DualWriterMode2) List(ctx context.Context, options *metainternalversion
 	return ll, nil
 }
 
-func enrichObject(orig, copy runtime.Object) (runtime.Object, error) {
-	accessorC, err := meta.Accessor(copy)
-	if err != nil {
-		return nil, err
-	}
-	accessorO, err := meta.Accessor(orig)
-	if err != nil {
-		return nil, err
-	}
-
+func enrichObject(accessorO, accessorC metav1.Object) {
 	accessorC.SetLabels(accessorO.GetLabels())
 
 	ac := accessorC.GetAnnotations()
 	for k, v := range accessorO.GetAnnotations() {
 		ac[k] = v
 	}
 	accessorC.SetAnnotations(ac)
+}
+
+// Update overrides the behavior of the generic DualWriter and writes first to LegacyStorage and then to Storage.
+func (d *DualWriterMode2) Update(ctx context.Context, name string, objInfo rest.UpdatedObjectInfo, createValidation rest.ValidateObjectFunc, updateValidation rest.ValidateObjectUpdateFunc, forceAllowCreate bool, options *metav1.UpdateOptions) (runtime.Object, bool, error) {
+	legacy, ok := d.Legacy.(rest.Updater)
+	if !ok {
+		return nil, false, errNoUpdateMethod
+	}
+
+	// get old and new (updated) object so they can be stored in legacy store
+	old, err := d.Storage.Get(ctx, name, &metav1.GetOptions{})
```

Comment:
We should be reading from Storage if possible and falling back to LegacyStorage if not. 

If we wait until https://github.com/grafana/grafana/pull/85968 is merged, we can directly call d.Get. The mode 2 DualWriter Get from that PR does all of that for us. It checks Storage and falls back to LegacyStorage if necessary.

Reply:
we can wait for that, yes :)